Google.Api.Generator/Utils/ResourcePattern.cs

```diff
@@ -52,7 +52,24 @@ public Segment(string segment)
                         {
                             throw new ArgumentException($"Segment '{segment}' is ill-formed; incorrect '}}' found.");
                         }
-                        parameterNames.Add(sb.ToString());
+                        var s = sb.ToString();
+                        if (s.EndsWith("=**"))
+                        {
+                            s = s[0..^3];
+                        }
+                        else if (s.EndsWith("=*"))
+                        {
+                            s = s[0..^2];
+                        }
+                        if (s == "")
+                        {
+                            throw new ArgumentException($"Segment '{segment}' is ill-formed; it is empty.");
```

Comment:
I suspect we may want to tighten up on terminology when we write the AIP. I tend to think of "segment" as a bit of the pattern between slashes. The bit in braces might be a "placeholder specifier" or similar.

So `~` is valid in segment, but not in a placeholder specifier.

But that can all wait - it's an implementation detail.

Reply:
Yes, agreed that various implemention details need tightening up.
I'm expecting to do this once the spec is finalized, as you suggest :) However, it's good enough for right now.